src/vs/workbench/parts/extensions/common/extensionsFileTemplate.ts

```diff
@@ -16,7 +16,16 @@ export const ExtensionsConfigurationSchema: IJSONSchema = {
 	properties: {
 		recommendations: {
 			type: 'array',
-			description: localize('app.extensions.json.recommendations', "List of extensions recommendations. The identifier of an extension is always '${publisher}.${name}'. For example: 'vscode.csharp'."),
+			description: localize('app.extensions.json.recommendations', "List of extensions which should be recommended for users of this repository. The identifier of an extension is always '${publisher}.${name}'. For example: 'vscode.csharp'."),
```

Comment:
Use `workpsace` instead of `repository` here and for `unwantedRecommendations`

This works for users not on git. So the use the word "repository" is not suggested.

"workspace" is more of a VS Code term which is more applicable